core/dbt/cli/main.py

```diff
@@ -73,6 +76,14 @@ def cli(ctx, **kwargs):
     if flags.RECORD_TIMING_INFO:
         ctx.with_resource(profiler(enable=True, outfile=flags.RECORD_TIMING_INFO))
 
+    project_root = flags.PROJECT_DIR if flags.PROJECT_DIR else os.getcwd()
+    cli_vars = parse_cli_vars(getattr(flags, "vars", "{}"))
+
+    # TODO need profile to exisit
+    profile = None
```

Comment:
@iknox-fa this dependency not removable, because of the need for target in profile. And due to the other comment we still need to pass in the whole profile object to load project

Reply:
I think we can integrate the logic in `parse_cli_vars` into the custom option type for YAML.  It's already doing 90% of this anyway, we just need to check that the yaml string evaluates to a dict and set a default value to `{}`.  If we do that we can just use `flags.vars` when calling `load_project`.

Reply:
Side note:  We'll want to support the "short form" of vars @jtcohen6 pointed out here: https://dbt-labs.slack.com/archives/C03P5DB4GA3/p1667422670997319

Reply:
Regarding the entire profile being a dependency of project:

I _really_ don't like that these two things are coupled right now.  Especially since the project requires the _entire_ profile.  I see what you're saying about how widely spread the renderer is across the codebase, but I'm going to take a little more time and see if I can find a way past it.

I won't hold up the PR on that account though-- We can always ticket decoupling the two as another piece of work.

Reply:
Explaining this to myself: `dbt_project.yml` can include Jinja expressions that use [`{{ target }}`](https://docs.getdbt.com/reference/dbt-jinja-functions/target), so we need to have loaded the profile before we can render those parts of `dbt_project.yml`.

_Another comment that's very much out of scope for this PR:_ We might also think about a future where one project/DAG could to be deployed using multiple adapters / targets. (They _could_ all be defined within the same profile, though.) At that point, we might especially want a clear decoupling between projects and connection profiles. The info arch around `target`, `profile`, `envs` (!?) is something we should look to refactor next year.

Reply:
Read my mind @jtcohen6 , I think I was writing somewhere that we should separate out target and the actual connection cred stuff.

@iknox-fa  I will also spend some more time try to figure out if we can get that rendered in some other form, so that maybe we just need to pass in target other than the whole object

Reply:
@iknox-fa I didn't do the

Reply:
Just to close the loop-- neither of us came up with a good way to break this apart without adding a lot to this ticket so we're leaving it as is.  https://github.com/dbt-labs/dbt-core/issues/6257 is the follow on ticket to decouple things.